front/src/components/UserVacationsComponent/ShowUserVacations.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useEffect, useState} from 'react';
+import {UserVacationsResponse, UserScheduleRequestApi} from "@noadudai/scheduler-backend-client/api";
+import axios from "axios";
+import {useAuth0} from "@auth0/auth0-react";
+import {useMutation} from "@tanstack/react-query";
+import { IoTrashOutline } from "react-icons/io5";
+import { HiOutlinePencil } from "react-icons/hi";
+import Pagination from "../Pagination.tsx";
+import {GetUserFutureVacationsPaginationModel} from "@noadudai/scheduler-backend-client";
+import GetPagesUserVacations from "./GetPagesUserVacations.tsx";
+
+
+const ShowUserVacations = () => {
+    const {getAccessTokenSilently} = useAuth0();
+
+    const [itemsPerPage] = useState(8);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [totalNumberOfVacations, setTotalNumberOfVacations] = useState(1);
+
+    const ax = axios.create({
+        baseURL: `${import.meta.env.VITE_BACKEND_BASE_URL}`,
+    });
+
+    const userSchedulePreferenceRequestApi: UserScheduleRequestApi = new UserScheduleRequestApi(undefined, undefined, ax);
+
+    const changePage = (pageNumber: number) => {
+        setCurrentPage(pageNumber)
+    };
+
+    const userSchedulePrefReqPostRequestNumberVacations = useMutation({
```

Comment:
extract this function together with the other api functions to a file called `apis.ts` and give them as hooks. the convention being `useX`  (`useQueryCurrentUserVacations`